### ISSUE TITLE
Certain inputs trigger infinite loop in `findFirstLineBreakOutsideComment`

### ISSUE THREAD

**Issue by Silic0nS0ldier** [rollup/rollup#4672]
### Rollup Version

v3.1.0

### Operating System (or Browser)

macOS

### Node Version (if applicable)

v16.17.1

### Link To Reproduction

https://github.com/Silic0nS0ldier/rollup-bug-20221014

### Expected Behaviour

Build completes

### Actual Behaviour

Build stalls, and never finishes.

**Comment by Silic0nS0ldier** [comment]
Issues appear to have occurred in this same area in the past. #3067

Some extra info.

Original source
```
#!/usr/bin/env node
import "source-map-support/register.js";
async function main() {
}
main();

```

What is passed into `findFirstLineBreakOutsideComment`
```
#!/usr/bin/env node

```

**Comment by Silic0nS0ldier** [comment]
It is possible to workaround this by changing the import, which presumably alters the string given to `findFirstLineBreakOutsideComment`.

e.g.
```
#!/usr/bin/env node
import * as _ from "source-map-support/register.js";
async function main() {
}
main();

```

**Comment by lukastaegert** [comment]
Thanks for the issue and the spot-on analysis! Fix at #4676.
